fem/coefficient.cpp

```diff
@@ -775,4 +775,93 @@ double ComputeGlobalLpNorm(double p, VectorCoefficient &coeff, ParMesh &pmesh,
 }
 #endif
 
+VectorQuadratureFunctionCoefficient::VectorQuadratureFunctionCoefficient(
+   QuadratureFunction *qf)
+   : VectorCoefficient(qf->GetVDim()), QuadF(qf), index(0),
+     length(qf->GetVDim()) {}
+
+void VectorQuadratureFunctionCoefficient::SetQuadratureFunction(
+   QuadratureFunction *qf)
+{
+   MFEM_VERIFY(QuadF, "QuadratureFunction must be set to a nonnull ptr");
+
+   index = 0;
+   length = qf->GetVDim();
+   vdim = length;
+   QuadF = qf;
+}
+
+void VectorQuadratureFunctionCoefficient::SetComponent(int _index, int _length)
+{
+   MFEM_VERIFY(_index >= 0, "Index must be >= 0");
+   MFEM_VERIFY(_index < QuadF->GetVDim(),
+               "Index must be < QuadratureFunction length");
+   index = _index;
+
+   MFEM_VERIFY(_length > 0, "Length must be > 0");
+
+   int diff = QuadF->GetVDim() - index;
+   MFEM_VERIFY(_length <= diff,
+               "Length must be <= (QuadratureFunction length - index)");
+
+   length = _length;
+   vdim = length;
```

Comment:
`length = vdim` seems to be forced always, so there's probably no need to have `length` as variable.

Reply:
fixed